### ISSUE TITLE
DETR VEA uses output of first layer of transformer decoder?

### ISSUE THREAD

**Issue by KamilDre** [tonyzhaozh/act#10]
In https://github.com/tonyzhaozh/act/blob/694c606940fc16289bafbd4c5d2c70bc27a98207/detr/models/detr_vae.py#L131 and https://github.com/tonyzhaozh/act/blob/694c606940fc16289bafbd4c5d2c70bc27a98207/detr/models/detr_vae.py#L136, you index the output of the transformer with [0]. Does this not take the output of the first layer of the transformer decoder, instead of the last layer? And is this behaviour expected?

**Comment by CarlDegio** [comment]
Agree. I removed subsequent layers and it had no effect on training.

**Comment by uuu686** [comment]
@CarlDegio hello,I changed 0 to -1 and used the output of the last layer, but there is no obvious difference in the effect. May I ask that your experiment is effective?

**Comment by CarlDegio** [comment]
I did not try the effect of multi-layer decoder. I just removed the decoder forward propagation that was not used in the original code to speed up the training. @uuu686
